cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java

```diff
@@ -227,50 +291,74 @@ private void upgradeCoProcessor(TableId tableId, Class<? extends Coprocessor> co
    * Creates a new instance of {@link HTable} for the given {@link TableId}. If the hbase table doesn't
    * exist, a new one will be created with the given number of splits.
    */
-  private HTable createTableIfNotExists(TableId tableId, int splits,
-                                        Class<? extends Coprocessor> coprocessor) throws IOException {
-    // Create the table if the table doesn't exist
-    try (HBaseAdmin admin = new HBaseAdmin(hConf)) {
-      // If table exists, then skip creating coprocessor etc
-      if (tableUtil.tableExists(admin, tableId)) {
-        return tableUtil.createHTable(hConf, tableId);
+  private HTableWithRowKeyDistributor createTable(TableId tableId, int splits,
+                                                  Class<? extends Coprocessor> coprocessor) throws IOException {
+
+    // Lookup the table descriptor from the cache first. If it is there, we assume the HBase table exists
+    // Otherwise, attempt to create it.
+    HTable hTable = null;
+    HTableDescriptor htd = tableDescriptors.get(tableId);
```

Comment:
yes. It is for getting the distributor bucket size:

https://github.com/caskdata/cdap/pull/7675/files/e6b438018362203cf5a68eabc82d60b3d95f4fcd#diff-9f9eef251f24002e8603485d1f069276R349